open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/impl/VRPConsentRequestHandler.java

```diff
@@ -0,0 +1,232 @@
+import com.wso2.openbanking.accelerator.consent.extensions.manage.validator.VRPConsentRequestValidator;
+import com.wso2.openbanking.accelerator.consent.extensions.util.ConsentManageUtil;
+import com.wso2.openbanking.accelerator.consent.mgt.dao.models.ConsentResource;
+import com.wso2.openbanking.accelerator.consent.mgt.dao.models.DetailedConsentResource;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.wso2.openbanking.accelerator.consent.extensions.common.ConsentExtensionConstants.AUTH_TYPE_AUTHORIZATION;
+import static com.wso2.openbanking.accelerator.consent.extensions.common.ConsentExtensionConstants.CREATED_STATUS;
+
+/**
+ * Consent Manage request handler class for VRP Payment Request Validation.
+ */
+public class VRPConsentRequestHandler implements ConsentManageRequestHandler {
+
+    private static final Log log = LogFactory.getLog(VRPConsentRequestHandler.class);
+
+
+    /**
+     * This method is responsible for processing a Variable Recurring Payment Consent Manage POST request.
+     * It validates the payment  request, checks for the existence of an idempotency key.
+     *
+     * @param consentManageData Object
+     */
+    @Override
+    public void handleConsentManagePost(ConsentManageData consentManageData) {
+
+        try {
+            Object request = consentManageData.getPayload();
+
+            JSONObject validationResponse = VRPConsentRequestValidator.validateVRPPayload(request);
+
+            if (!(Boolean.parseBoolean(validationResponse.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                log.error(validationResponse.get(ConsentExtensionConstants.ERRORS));
+                throw new ConsentException((ResponseStatus) validationResponse
+                        .get(ConsentExtensionConstants.HTTP_CODE),
+                        String.valueOf(validationResponse.get(ConsentExtensionConstants.ERRORS)));
+            }
+
+            if (StringUtils.isEmpty(consentManageData.getHeaders()
+                    .get(ConsentExtensionConstants.X_IDEMPOTENCY_KEY))) {
+                log.error(ErrorConstants.IDEMPOTENCY_KEY_NOT_FOUND);
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.IDEMPOTENCY_KEY_NOT_FOUND);
+            }
+
+            //Handle payment initiation flows
+            handlePaymentPost(consentManageData, request);
+
+        } catch (ConsentManagementException e) {
+            log.error("Error occurred while handling the initiation request", e);
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                    ErrorConstants.PAYMENT_INITIATION_HANDLE_ERROR);
```

Comment:
`PAYMENT_INITIATION_HANDLE_ERROR` shouldn't this be VRP initiation error?

Reply:
Fixed in [876c68f]